rust/connlib/tunnel/src/tests/composite_strategy.rs

```diff
@@ -0,0 +1,104 @@
+use proptest::strategy::{BoxedStrategy, Strategy, Union};
+use std::fmt;
+
+#[derive(Debug)]
+pub struct CompositeStrategy<T>
```

Comment:
I think I know why this is called `CompositeStrategy` from our chat yesterday but a comment would be nice